framework/db/src/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -969,7 +969,12 @@ public T findByUuidIncludingRemoved(final String uuid) {
     @Override
     @DB()
     public T findByIdIncludingRemoved(ID id) {
-        return findById(id, true, null);
+        if (_cache != null) {
+            final Element element = _cache.get(id);
+            return element == null ? findById(id, true, null) : (T)element.getObjectValue();
+        } else {
```

Comment:
@DaanHoogland following code returns null if _cache is not null but element is null,
>`
T result = null
if (_cache != null) {
    final Element element = _cache.get(id);
    if (element != null) {
        result = (T)element.getObjectValue();
    }
} else {
    result =findById(id, true, null);
}
return result;
`